src/test/java/teammates/ui/webapi/GetActionClassesActionTest.java

```diff
@@ -0,0 +1,47 @@
+package teammates.ui.webapi;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.testng.annotations.Test;
+
+import teammates.common.util.Const;
+import teammates.ui.output.ActionClasses;
+
+/**
+ * SUT: {@link GetActionClassesAction}.
+ */
+public class GetActionClassesActionTest extends BaseActionTest<GetActionClassesAction> {
+
+    @Override
+    protected String getActionUri() {
+        return Const.ResourceURIs.ACTION_CLASS;
+    }
+
+    @Override
+    protected String getRequestMethod() {
+        return GET;
+    }
+
+    @Test
+    @Override
+    protected void testExecute() throws Exception {
+        ______TS("Retrieve the list of action class names");
+        GetActionClassesAction action = getAction();
+        action.execute();
+        JsonResult result = getJsonResult(action);
+        ActionClasses data = (ActionClasses) result.getOutput();
+        List<String> expectedActionClasses = ActionFactory.ACTION_MAPPINGS.values().stream()
```

Comment:
While this is a white-box test, this is going a bit too far. This is literally copying the implementation and passing it as the test. Where is the check if either side (if there are even two sides, because right now both are the same) does not do the job?

Reply:
Understood. To make the changes, I have created a list of all action classes names to be compared with the action output. For extension, more action class names can just be added to the end of the list.